src/test/kotlin/no/nav/syfo/oppfolgingstilfelle/retry/KafkaOppfolgingstilfelleRetrySpek.kt

```diff
@@ -0,0 +1,367 @@
+
+                runBlocking {
+                    pollAndProcessOppfolgingstilfelleRetryTopic(
+                        kafkaConsumer = mockConsumerOppfolgingstilfelleRetry,
+                        oppfolgingstilfelleRetryService = oppfolgingstilfelleRetryService
+                    )
+                }
+
+                val messages = getMessagesOversikthendelsetilfelle(consumerOversikthendelsetilfelle)
+                messages.size shouldBeEqualTo 1
+                messages.first().fnr shouldBeEqualTo ARBEIDSTAKER_FNR.value
+                messages.first().enhetId shouldBeEqualTo behandlendeEnhetMock.behandlendeEnhet.enhetId
+                messages.first().virksomhetsnummer shouldBeEqualTo kOppfolgingstilfelleRetry.orgnummer
+            }
+
+            it("should skip sending of Oversikthendelsetilfelle when retry has exceeded retry limit") {
+                val kOppfolgingstilfelleRetry = generateKOppfolgingstilfelleRetry.copy(
+                    created = LocalDateTime.now().minusHours(RETRY_OPPFOLGINGSTILFELLE_INTERVAL_MINUTES).minusMinutes(1),
+                    retriedCount = RETRY_OPPFOLGINGSTILFELLE_COUNT_LIMIT
+                )
+                val mockConsumerOppfolgingstilfelleRetry = createMockConsumerOppfolgingstilfelleRetry(kOppfolgingstilfelleRetry)
+
+                runBlocking {
+                    pollAndProcessOppfolgingstilfelleRetryTopic(
+                        kafkaConsumer = mockConsumerOppfolgingstilfelleRetry,
+                        oppfolgingstilfelleRetryService = oppfolgingstilfelleRetryService
+                    )
+                }
+
+                val messages = getMessagesOversikthendelsetilfelle(consumerOversikthendelsetilfelle)
+                messages.size shouldBeEqualTo 0
+            }
+        }
+
+        describe("Resend KOversikthendelseRetry") {
+            val mockOppfolgingstilfelleRetryProducer = mockk<OppfolgingstilfelleRetryProducer>()
+            justRun { mockOppfolgingstilfelleRetryProducer.sendAgainOppfolgingstilfelleRetry(any()) }
+            justRun { mockOppfolgingstilfelleRetryProducer.sendRetriedOppfolgingstilfelleRetry(any()) }
+
+            val oppfolgingstilfelleRetryService = OppfolgingstilfelleRetryService(
+                oppfolgingstilfelleService = oppfolgingstilfelleService,
+                oppfolgingstilfelleRetryProducer = mockOppfolgingstilfelleRetryProducer
+            )
+
+            val consumerPropertiesOppfolgingstilfelleRetry = kafkaOppfolgingstilfelleRetryConsumerProperties(env, vaultSecrets)
+                .overrideForTest()
+
+            val consumerOppfolgingstilfelleRetry = KafkaConsumer<String, String>(consumerPropertiesOppfolgingstilfelleRetry)
+            consumerOppfolgingstilfelleRetry.subscribe(listOf(oppfolgingstilfelleRetryTopic))
+
+            it("should resend KOversikthendelseRetry to topic when it has not exceeded retry limit and is not ready for retry") {
```

Comment:
Jeg bruker ofte tester for å skjønne hva koden gjør. Blir testen litt tydeligere hvis du legger inn ordet "vente", og evt hvorfor vi venter? Har jeg i så fall forstått det riktig at vi her tester at vi venter med å prosessere meldingen i tilfelle feilen retter seg selv, og vi venter ved å legge meldinga tilbake på topic?

Reply:
Lov å være uenig her altså!

Reply:
Denne sjekker at man sender elementet inn i "vaskemaskinen" igjen, altså inn i den evige loopen. Så en slags venting på at retryTime er basert, men venter ikke med å legge den tilbake på topic.
Så koden i testenventer ikke om det var det du mente(?)